lib-multisrc/heancms/src/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt

```
@@ -70,6 +76,39 @@ abstract class HeanCms(
         .add("Origin", baseUrl)
         .add("Referer", "$baseUrl/")
 
+    private fun authHeaders(): Headers {
+        val builder = headersBuilder()
+        if (preferences.loginPref && enableLogin) {
+            val tokenData = preferences.tokenData
+            val token = if (tokenData.isExpired(tokenExpiredAtDateFormat)) {
+                getToken()
+            } else {
+                tokenData.token
+            }
+            if (token != null) {
+                builder.add("Authorization", "Bearer $token")
+            }
+        }
+        return builder.build()
+    }
+
+    private fun getToken(): String? {
+        val body = FormBody.Builder()
+            .add("email", preferences.user)
+            .add("password", preferences.password)
+            .build()
+
+        val response = client.newCall(POST("$apiUrl/login", headers, body)).execute()
+
+        if (!response.isSuccessful) return null
```

maybe throw if bad request like invalid username or password